rust/rope/src/engine.rs

```diff
@@ -110,7 +119,7 @@ impl Engine {
 
     /// Get text of head revision.
     pub fn get_head(&self) -> Rope {
-        self.rev_content_for_index(self.revs.len() - 1)
+        self.text.clone()
```

Comment:
Isn't `Rope::clone()` super cheap? I assumed it was basically just an `Arc::clone()` but I admit I didn't actually check. But even if that's the case it probably would make more sense to let the user do that if necessary.